crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -850,8 +851,51 @@ pub struct RoomInfo {
     pub(crate) base_info: Box<BaseRoomInfo>,
 }
 
+/// This is the same as RoomInfo, except it contains RawBaseRoomInfo and implements Serialize.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RawRoomInfo {
```

Comment:
The sheer amount of code duplication scares me, here; is there a way to have an intermediate struct `CommonRoomInfo<T>`, which would implement `Serialize`, and then `RawRoomInfo` would be `CommonRoomInfo<RawBaseRoomInfo>` while `RoomInfo` would be `CommonRoomInfo<BaseRoomInfo>`?